app/Services/InviteService.php

```diff
@@ -2,44 +2,164 @@
             ]);
         }
         $invite->load('roles');
+        $this->sendMail($invite);
         return $invite;
     }
-}
\ No newline at end of file
+
+    public function resolve($id)
+    {
+        $invite = $this->repo()->single($id);
+        $roles = $invite->roles()->get();
+        $user = auth()->user();
+        $school_id = $invite->school_id;
+
+        foreach ($roles as $_role) {
+            $role_id = $_role->role_id;
+            $extras = json_decode($_role->extras);
+            $role = $this->roleRepository->single($role_id);
+
+            if ($role->name == Role::ADMIN) {
+                $admin = $user->roles()->attach([
+                    $role->id => [
+                        'school_id' => $school_id,
+                    ],
+                ]);
+                return $admin;
+            }
+
+            if ($role->name == Role::SCHOOL_OWNER) {
+                $staff = Role::whereName(Role::STAFF)->first();
+
+                $user->roles()->attach([
+                    $role->id => [
+                        'school_id' => $school_id,
+                    ],
+                    $staff->id => [ //create staff role
+                        'school_id' => $school_id,
+                    ],
+                ]);
+
+                //add as staff
+                $department_id = $extras ? $extras->department_id : null;
+                $opts = ['user_id' => $user->id, 'school_id' => $school_id, 'title' => '', 'department_id' => $department_id];
+                $staff = $this->staffRepository->create($opts);
+                return $staff;
+            }
+
+            if ($role->name == Role::STAFF) {
+                $user->roles()->attach([
+                    $role->id => [
+                        'school_id' => $school_id,
+                    ],
+                ]);
+
+                //create staff record
+                $department_id = $extras ? $extras->department_id : null;
+                $opts = ['user_id' => $user->id, 'school_id' => $school_id, 'title' => ''];
+                $staff = $this->staffRepository->create($opts);
+
+                if ($department_id) {
+                    $opts = ['department_id' => $department_id];
+                    $staff = $this->staffRepository->update($staff->id, $opts);
+                }
+                return $staff;
+
+            }
+
+            if ($role->name == Role::STUDENT) {
+                $user->roles()->attach([
+                    $role->id => [
+                        'school_id' => $school_id,
+                    ],
+                ]);
+
+                //create student record
+                $program_id = $extras ? $extras->program_id : null;
+                $matric_no = $extras ? $extras->matric_no : null;
+                $opts = ['user_id' => $user->id, 'matric_no' => $matric_no, 'program_id' => $program_id];
+                $student = $this->studentRepository->create($opts);
+
+                return $student;
```

Comment:
You should always return the user. If you need to include the student's details, you could use:

```php
$user['student'] = $student;
return $user;
```